File: HackRank-DataStructure/Array and Linked List/Dynamic Array.py
def dynamicArray(n, queries):
    v = []
    for i in range(n):
        v.append([])
    ans = []
    last = 0
    for q in queries:
        idx = (q[1] ^ last) % n
        if(q[0] == 1):
            v[idx].append(q[2])
        else:
            last = v[idx][q[2] % len(v[idx])]
            ans.append(last)
    return ans

File: HackRank-DataStructure/Array and Linked List/Dynamic Array.cpp
vector<int> dynamicArray(int n, vector<vector<int>> queries) {
    vector<vector<int>> v;
    for(int i = 0;i < n;i++) v.push_back(*(new vector<int>()));
    vector<int> ans;
    int last = 0;
    for(vector<int>vv: queries)
    {
        int idx = (vv[1] ^ last) % n;
        if(vv[0] == 1)
        {
            v[idx].push_back(vv[2]);
        }
        else
          {
              last = v[idx][vv[2] % v[idx].size()];
              ans.push_back(last);
        }
    }
    return ans;
}